### ISSUE TITLE
Multiple Detections causing problems

### ISSUE THREAD

**Issue by adplant** [juliansteenbakker/mobile_scanner#358]
I think this may be a bug. If not, feel free to close, but it's not clear to me why a detected barcode wouldn't finish the onDetect logic prior to scanning another barcode.

### Discussed in https://github.com/juliansteenbakker/mobile_scanner/discussions/357

<div type='discussions-op-text'>

<sup>Originally posted by **adplant** November  2, 2022</sup>
I've got an AlertDialog that I include a button to scan a MobileScanner() from. It works great most of the time, but I've noticed on one barcode in particular that onDetect gets called multiple times before popping the screen, and I don't know why.

This is the code:

```
                            MobileScanner(
                              onDetect: (barcode, args) async {
                                if (barcode.rawValue == null) {
                                  debugPrint('Failed to scan Barcode');
                                  Navigator.pop(context);
                                } else {
                                  final String code = barcode.rawValue!;
                                  debugPrint(
                                      'Barcode found! $code type: ${barcode.type}');
                                  textController.text = code;
                                  Navigator.pop(context);
                                }
                              },
                            ),
```
And my console ends up looking something like this (notice the two different results that are found in the rawValue):

```
I/flutter (32327): Barcode found! OIIPPP6 7 type: BarcodeType.text
I/flutter (32327): Barcode found! XIIPPP6 P type: BarcodeType.text
E/flutter (32327): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/widgets/navigator.dart': Failed assertion: line 4959 pos 16: 'entry.route._popCompleter.isCompleted': is not true.
E/flutter (32327): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter (32327): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter (32327): #2      NavigatorState.pop (package:flutter/src/widgets/navigator.dart:4959:16)
E/flutter (32327): #3      Navigator.pop (package:flutter/src/widgets/navigator.dart:2433:27)
.....
.....
E/flutter (32327): #6      _RootZone.runUnaryGuarded (dart:async/zone.dart:1586:10)
E/flutter (32327): #7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter (32327): #8      _DelayedData.perform (dart:async/stream_impl.dart:515:14)
E/flutter (32327): #9      _PendingEvents.handleNext (dart:async/stream_impl.dart:620:11)
E/flutter (32327): #10     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:591:7)
E/flutter (32327): #11     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
E/flutter (32327): #12     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
```

So basically, I think I just need to know how to make sure the MobileScanner only detects one time before popping the screen. I do, however, need to still be able to select the Scan button as many times as I want while using the app and have the Scanner page populate.</div>

**Comment by parmarjitesh** [comment]
Same Issue For Me. Any Help will be Highly Appreciated

**Comment by robsoncardozorebel** [comment]
The same issue here with mobile_scanner 3.0.0-beta.1

**Comment by Repox** [comment]
I also have this issue at the moment.

This messes up navigation.

**Comment by sachinbpatil619** [comment]
did they removed allowDuplicates property ?
or any suggestions to avoid multiple scan ?

**Comment by adplant** [comment]
@sachinbpatil619 No. I'm using the allowDuplicates: false, but that's not the same thing that this tickets is talking about. Notice the logs. The scanner is scanning multiple times very quickly and getting a DIFFERENT result each time. They're not duplicates. We need a way to only scan once. I thought the code was written to make that the case, but multiple scans keep happening for certain bar/qr codes.

**Comment by juliansteenbakker** [comment]
Please check #362 , i've recreated the detection function so this issue will probably be solved with it.

**Comment by adplant** [comment]
@juliansteenbakker Yeah, I think you're probably right. Thanks a lot. Do you think you'll finish merging soon? I look forward to trying it out.

**Comment by juliansteenbakker** [comment]
This is now available in version 3.0.0-beta.4.

**Comment by zanovis** [comment]
> You can now control the DetectionSpeed, as well as the timeout of the DetectionSpeed. For more info see the DetectionSpeed documentation. This replaces the allowDuplicates function.

Where is the DetectionSpeed documentation? Thank you!

**Comment by zanovis** [comment]
> This is now available in version 3.0.0-beta.4.

Just implemented 3.0.0-beta, it is working well. Thank you for your work on this!

**Comment by kamranibrahim** [comment]
is there any way to know that duplicate has occur

**Comment by georges-ph** [comment]
version `3.5.5` and still happening
My solution for it was to dispose the controller before `Navigator.pop(context)`

**Comment by LahiruuChanaka** [comment]
I am experiencing the same issue. Despite it being reported over a year ago, it remains unresolved. Specifically, the onDetect method is being triggered multiple times after a barcode is detected.
